flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/sessionjob/FlinkSessionJobReconciler.java

```diff
@@ -83,6 +84,13 @@ public void reconcile(FlinkSessionJob flinkSessionJob, Context context) throws E
 
         Configuration deployedConfig = configManager.getObserveConfig(flinkDepOptional.get());
 
+        // merge session job specific config
+        Map<String, String> sessionJobFlinkConfiguration =
```

Comment:
nit: maybe we could move this to the `FlinkConfigManager`

Reply:
I created a `getSessionJobObserveConfig` method in `FlinkConfigManager` and replace this one and a few other places.